sympy/core/add.py

```diff
@@ -576,6 +580,8 @@ def _eval_is_positive(self):
             v = _monotonic_sign(a)
             if v is not None:
                 s = v + c
+                if s == self:
+                    return None
                 if s.is_positive and a.is_nonnegative:
```

Comment:
I would just be parroting back the docstring: the basic idea is to return something indicative of the sign when all symbols are given a signed value/symbol consistent with their assumptions. If s is the same as self it means that the symbol-containing part came back unchanged from `_monotonic_sign`. What sort of expression is giving a recursion error/problem?